vss-extension.json

```diff
@@ -117,42 +117,76 @@
                     "displayName": "Marketplace URL",
                     "value": "https://marketplace.visualstudio.com"
                 },
-                "authenticationSchemes": [
+              "authenticationSchemes": [
+                {
+                  "type": "ms.vss-endpoint.endpoint-auth-scheme-token",
```

Comment:
I needed to revisit this auth-scheme internally to remember why it wasn't used in the first place. The `endpoint-auth-scheme-token` is configured to add an Authorization header with just the token that the user fills in. If we had dataSources in this endpoint-type, then the HTTP calls to the gallery would end with an auth header like this `Authorization: {apitoken}`. So, no authorization scheme is being specified (Basic, Bearer, etc.), which wouldn't work in VSTS/TFS. That's the reason why we used Basic Auth from the start, which works with dataSources even when using PATs with basic auth (in fact `tfx-cli` uses also basic auth internally even when `--token` is passed). 

I'm not sure if a service endpoint is allowed to be saved with an empty Url, but if it is, I would suggest to keep the basic-auth, to stop defaulting to the marketplace Url in the endpoint type to make it visible, make also visible the Username and set it as optional, to change the label of PAT to Password/PAT, and then in code, if no Url defined, we could assume the public gallery. I just need to confirm if it's allowed to save a service endpoint with an empty URL. Probably it's not.